src/i965_gpe_utils.c

```diff
@@ -1394,7 +1394,12 @@ i965_object_surface_to_2d_gpe_resource_with_align(struct i965_gpe_resource *res,
     unsigned int swizzle;
 
     res->type = I965_GPE_RESOURCE_2D;
-    res->width = ALIGN(obj_surface->orig_width, (1 << alignment));
+
+    if (obj_surface->fourcc == VA_FOURCC_P010)
+        res->width = ALIGN(obj_surface->orig_width * 2, (1 << alignment));
+    else
+        res->width = ALIGN(obj_surface->orig_width, (1 << alignment));
+
```

Comment:
it should be based on the kernel implementation. it always need to set the format as 8bits even for P010